.circleci/config.yml

```diff
@@ -92,8 +92,10 @@ jobs:
       - run: |
           # NOTE: -Werror disabled in CI, as we currently have >4k warnings.
           cd build \
+            && apt-get update && apt-get install --assume-yes lld \
             && cmake .. -DCMAKE_BUILD_TYPE=Debug -DENABLE_COVERAGE=On -DCPACK_GENERATOR=DEB \
                 -DENABLE_COMPILER_WARNINGS=On -DENABLE_WERROR=Off -DCMAKE_EXPORT_COMPILE_COMMANDS=On \
+                -DCMAKE_CXX_FLAGS="-fuse-ld=lld" \
```

Comment:
should we default to this inside of cmake if we detect its presence, also why not the other builds?

Reply:
It seemed far from obvious how to do it inside `cmake` except for manually enabling it for every single target. [Stackoverflow thread](https://stackoverflow.com/questions/1867745/cmake-use-a-custom-linker)

 A workaround was to use this compiler flag which works if you haven't already made specific linker configuration inside cmake

Reply:
For some reason I wasn't able to see a speedup when testing with lld locally.

Without `lld`:
```shell
$ cmake .. -DCMAKE_BUILD_TYPE=RelWithDebInfo -DLOGGING_LEVEL=ERROR -DENABLE_NODE_BINDINGS=Off -DENABLE_PYTHON_BINDINGS=Off -DCMAKE_CXX_COMPILER_LAUNCHER=ccache -DBUILD_SHARED_LIBS=On
...
$ make clean
$ time make -j(nproc) check
...
Executed in   67.45 secs   fish           external 
   usr time  267.32 secs  1355.00 micros  267.32 secs 
   sys time   52.82 secs  397.00 micros   52.82 secs
```

With `lld`:
```shell
$ cmake .. -DCMAKE_BUILD_TYPE=RelWithDebInfo -DLOGGING_LEVEL=ERROR -DENABLE_NODE_BINDINGS=Off -DENABLE_PYTHON_BINDINGS=Off -DCMAKE_CXX_COMPILER_LAUNCHER=ccache -DBUILD_SHARED_LIBS=On -DCMAKE_CXX_FLAGS="-fuse-ld=lld"
...
$ make clean
$ time make -j(nproc) check
Executed in   67.14 secs   fish           external 
   usr time  266.98 secs  1792.00 micros  266.98 secs 
   sys time   52.11 secs    0.00 micros   52.11 secs 
```

Reply:
ok, so the reason I dont see a speedup is because I have `-DBUILD_SHARED_LIBS=On` which builds a shared lib (and does not have as much a hit on link as a static lib).

@PureW has confirmed that speedup with static lib is significant. Lets roll this change! :+1:

Reply:
@PureW I've had trouble using `lld` to link valhalla as a **shared lib** after the changes introduced in https://github.com/valhalla/valhalla/pull/2787 (I can't see any issue with the PR itself though, but the link failure happens after this change was introduced).

Can you try it on your setup and see if valhalla links successfully with `-DBUILD_SHARED_LIBS=On`?

Reply:
@mandeepsandhu simply from turning on `BUILD_SHARED_LIBS` gives me linking errors with geos which seem to be caused by cmake not passing geos to the linker command AFAICT

> FAILED: valhalla_add_predicted_traffic   
> : && /usr/bin/c++ -fuse-ld=lld -fdiagnostics-color=auto  -g  CMakeFiles/valhalla_add_predicted_traffic.dir/src/mjolnir/valhalla_add_predicted_traffic.cc.o -o valhalla_add_predicted_traffic  -Wl,-rpath,/home/anders/Documents/Programming/mapbox/valhalla/build-clang/src:  src/libvalhalla.so.3.1.0  /usr/lib/libboost_program_options.so.1.75.0  /usr/lib/libprotobuf-lite.so  /usr/lib/libcurl.so  /usr/lib/libz.so  /usr/lib/libprime_server.so  /usr/lib/libspatialite.so  /usr/lib/libsqlite3.so  /usr/lib/libluajit-5.1.so  -lm  -pthread && :
> ld.lld: error: src/libvalhalla.so.3.1.0: undefined reference to geos::geom::GeometryFactory::destroy() [--no-allow-shlib-undefined]

 :disappointed: 

I began this work to bring down the test-times initially. Maybe we should instead simply turn on `BUILD_SHARED_LIBS` and stay with `ld` for the tests as shared libs avoid the expensive static link time.

Reply:
yeah I assume it has something to do with how we're [importing libgeos into cmake](https://github.com/valhalla/valhalla/blob/master/CMakeLists.txt#L308). Maybe `lld` doesn't like the generated link command for it. Yeah switching to shared lib for tests seems like a low-hanging fruit.